### PR TITLE
Update magnetic bottle demo

### PR DESCRIPTION
**Pull request by henry2004y**
* Use Makie instead of PyPlot
* Change initial energy due to the fix in relativistic tracing
* Separate magnetic mirror configuration from magnetic bottle

### CODE REVIEW

File: examples/demo_magneticbottle.jl [hunk 1]
```diff
@@ -51,51 +50,85 @@ stateinit = [r₀..., v₀...]
...
+
+v_str = @sprintf "Speed = %4.2f %s" vratio*100 "% speed of light"
+e_str = @sprintf "Energy = %6.4f MeV" E
+
+println(v_str)
+println(e_str)
 
 # Default Tsit5() and many solvers does not work in this case!
-sol = solve(prob, AB3(); dt=1.5e-12, save_idxs=[1,2,3])
+sol_rel = solve(prob_rel, AB3(); dt=1.5e-12, save_idxs=[1,2,3])
```
Comment (TCLiuu) [inline examples/demo_magneticbottle.jl:66]: Why do we need to use `AB3` here? I have encountered similar problems many times when calculating relativistic situations. Is there anything special about this algorithm, or did you discover it through trial and error?

Comment (henry2004y) [reply examples/demo_magneticbottle.jl:66]: This was based on previous trials (on the wrong relativistic equations). In principle AB3 should be better for stiff problems, of which relativistic Lorentz equation may be. However, the applicability of non-stiff solvers like Tsit5 should be checked again together with other quantities like energy conservation.

**Comment by TCLiuu** [comment]
When I further tested this example, I discovered a rather peculiar phenomenon: it does not satisfy magnetic moment invariance. In the region where particles are approaching a turn, the magnetic field changes very quickly to the point that it does not meet the adiabatic invariant potential's slow change condition. This causes a drastic change in magnetic moment within this region.

**Comment by henry2004y** [comment]
Not sure if it is valid or not: should the magnetic moment be conserved?

**Pull request merged**